File: S2-Regression/Pt1-SimpleLinearRegression/simple_linear_regression.py
## Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


## Importing the datase
dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values


## Splitting the dataset into the training set and test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=.2, random_state=0)


## Traininf the Simple Linear Regression model on the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)


## Predicting the test results
y_pred = regressor.predict(x_test)


## Visualizing the training set
plt.scatter(x_train, y_train, color='blue')
plt.plot(x_train, regressor.predict(x_train), color='red')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()


## Visualizing the test set
plt.scatter(x_test, y_test, color='lime')
plt.plot(x_train, regressor.predict(x_train), color='red')
plt.title('Salary vs Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: S2-Regression/Pt1-SimpleLinearRegression/simple_linear_regression.R
## Importing the dataset
dataset = read.csv('Salary_Data.csv')

## Splitting the dataset into the training set and test set
# install.packages('caTools')
library('caTools')
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


## Fitting Simple Linear Regression into training set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)


## Predicting the test set results
y_pred = predict(regressor, newdata = test_set)


## Visualizing the training set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(
    x = training_set$YearsExperience, 
    y = training_set$Salary
  ), colour = 'blue') +
  geom_line(aes(
    x = training_set$YearsExperience, 
    y = predict(regressor, newdata = training_set)
  ), colour = 'red') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')
  

## Visualizing the test set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(
    x = test_set$YearsExperience,
    y = test_set$Salary
  ), colour = 'blue') +
  geom_line(aes(
    x = training_set$YearsExperience,
    y = predict(regressor, newdata = training_set)
  ), colour = 'red') +
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')
